protected/humhub/modules/file/models/File.php

```diff
@@ -8,56 +8,86 @@
 
 namespace humhub\modules\file\models;
 
+use humhub\components\ActiveRecord;
 use humhub\components\behaviors\GUID;
 use humhub\components\behaviors\PolymorphicRelation;
-use humhub\components\ActiveRecord;
 use humhub\modules\content\components\ContentActiveRecord;
 use humhub\modules\content\components\ContentAddonActiveRecord;
+use humhub\modules\file\components\StorageManager;
+use humhub\modules\file\components\StorageManagerInterface;
 use humhub\modules\user\models\User;
+use Throwable;
 use Yii;
 use yii\base\InvalidArgumentException;
+use yii\base\InvalidConfigException;
+use yii\db\ActiveQuery;
 use yii\db\Exception;
+use yii\db\IntegrityException;
+use yii\db\StaleObjectException;
 use yii\helpers\Url;
 use yii\web\UploadedFile;
 
 /**
  * This is the model class for table "file".
  *
- * The followings are the available columns in table 'file':
+ * The following are the available columns in table 'file':
+ *
  * @property integer $id
  * @property string $guid
+ * @property integer|null $category
  * @property string $file_name
  * @property string $title
  * @property string $mime_type
  * @property string $size
- * @property string $object_model
- * @property integer $object_id
- * @property integer $content_id
- * @property string $created_at
- * @property integer $created_by
- * @property string $updated_at
- * @property integer $updated_by
- * @property integer $show_in_stream
- * @property string $hash_sha1
+ * @property string|null $object_model
+ * @property integer|null $object_id
+ * @property integer|null $content_id
+ * @property string|null $created_at
+ * @property integer|null $created_by
+ * @property string|null $updated_at
+ * @property integer|null $updated_by
+ * @property integer|null $show_in_stream
+ * @property string|null $hash_sha1
+ * @property mixed|null $onlyoffice_key
  *
- * @property \humhub\modules\user\models\User $createdBy
- * @property \humhub\modules\file\components\StorageManager $store
+ * @property StorageManager $store
  * @property FileHistory[] $historyFiles
  *
  * @mixin PolymorphicRelation
  * @mixin GUID
  *
- * Following properties are optional and for module depended use:
+ * Following properties are optional and for module-dependent use:
  * - title
  *
  * @since 0.5
+ * @noinspection PropertiesInspection
  */
 class File extends FileCompat
 {
+    /**
+     * Categories are still experimental. Expect changes in v1.16
```

Comment:
This comment  is very helpful in understanding where this should be going. But I would leave out the constants in the beta for now. e.g. VARIANT_1 - 8 is quite unclear for me.

Reply:
Yes, the variants would need some explanation/documentation.

The idea was, that if we want to do something along the lines of bit-mask, then it was the idea to reserve the first 4 bits for variants of a given category.

E.e. if you have
- `00010000` = 16: Category A, you can have the following variants:
  - `10010000` = 17: Category A.x
  - `01010000` = 18: Category A.y
  - `11010000` = 19: Category A.z
  - ...
- `00001000` = 32: Category B, you can have the following variants:
  - `10001000` = 32: Category A.x
  - `01001000` = 34: Category A.y
  - `11001000` = 35: Category A.z
  - ...

Like so, the categories are close to each other in the decimal notation, and still usable with bit-mask.

PS: removed.